src/QtBase/Networking/WebsocketServer.cpp

```diff
@@ -0,0 +1,413 @@
+   return;
+   }
+
+   qDebug() << "Received WSMessage " << mMessageRecv.type() << " from client " << pClient;
+
+   // Run according handler
+   if (0 == ::strcmp(mMessageRecv.type(), WSMessage::Type::ON_URL_OPENED))
+   {
+      WSMessageOnUrlOpened& msg = *(WSMessageOnUrlOpened*)&mMessageRecv;
+      handleOnUrlOpened(msg.getUrl());
+   }
+   else if (0 == ::strcmp(mMessageRecv.type(), WSMessage::Type::ON_BROWSER_PLAYER_STARTED))
+   {
+      handleOnBrowserPlayerStarted();
+   }
+   else if (0 == ::strcmp(mMessageRecv.type(), WSMessage::Type::ON_BROWSER_PLAYER_STOPPED))
+   {
+      handleOnBrowserPlayerStopped();
+   }
+   else if (0 == ::strcmp(mMessageRecv.type(), WSMessage::Type::ON_BROWSER_PLAYER_PAUSED))
+   {
+      handleOnBrowserPlayerPaused();
+   }
+   else
+      qDebug() << "Unhandled WSMessage: " << mMessageRecv.type();
+}
+
+void WebsocketServer::processBinaryMessage(QByteArray message)
+{
+   QWebSocket* pClient = qobject_cast<QWebSocket*>(sender());
+   qDebug() << "Binary Message received:" << message;
+}
+
+// ENGINE EVENTS
+
+void WebsocketServer::OnStartSession()
+{
+   if (Session* s = GetSession())
+   {
+      mMessageOnSessionStarted.setSession(s);
+      sendMessage(mMessageOnSessionStarted);
+
+      mMessageOnFeedbackLayout.setClassifier(GetEngine()->GetActiveClassifier());
+      sendMessage(mMessageOnFeedbackLayout);
+      mTimerFeedbacks->start(100);
+   }
+}
+void WebsocketServer::OnStopSession()
+{
+   sendMessage(mMessageOnSessionStopped);
+}
+void WebsocketServer::OnActiveClassifierChanged(Classifier* classifier)
+{
+   if (!classifier)
+      return;
+   mMessageOnClassifierLoaded.setClassifier(classifier);
+   sendMessage(mMessageOnClassifierLoaded);
+}
+void WebsocketServer::OnActiveStateMachineChanged(StateMachine* stateMachine)
+{
+   if (!stateMachine)
+      return;
+   mMessageOnStateMachineLoaded.setStateMachine(stateMachine);
+   sendMessage(mMessageOnStateMachineLoaded);
+}
+void WebsocketServer::OnActiveExperienceChanged(Experience* experience)
+{
+   if (!experience)
+      return;
+   mMessageOnExperienceLoaded.setExperience(experience);
+   sendMessage(mMessageOnExperienceLoaded);
+}
+
+void WebsocketServer::OnNodeAdded(Graph* graph, Node* node)
+{
+   if (graph->GetType() == Classifier::TYPE_ID)
+   {
+   }
+}
+
+void WebsocketServer::OnRemoveNode(Graph* graph, Node* node)
+{
+   if (graph->GetType() == Classifier::TYPE_ID)
+   {
+   }
+}
+
+void WebsocketServer::OnConnectionAdded(Graph* graph, Connection* connection)
+{
+   if (graph->GetType() == Classifier::TYPE_ID)
+   {
+   }
+}
+
+void WebsocketServer::OnRemoveConnection(Graph* graph, Connection* connection)
+{
+   if (graph->GetType() == Classifier::TYPE_ID)
+   {
+   }
+}
+
+void WebsocketServer::OnOpenUrl(const char* url)
+{
+   mMessageOpenUrl.setUrl(url);
+   sendMessage(mMessageOpenUrl);
+}
+
+void WebsocketServer::OnBrowserStartPlayer(double progress, bool fullscreen)
+{
+   mMessageBrowserStartPlayer.setProgress(progress);

```

Comment:
It makes probably more sense to navigate to a certain time point in the video instead of the relative progress wrt the duration of the video. 
Also, can we make the progress parameter optional? Then we can also resume the video using the same action.